src/rules/selector-attribute-square-brackets-space-inside/index.js

```diff
@@ -0,0 +1,81 @@
+import selectorParser from "postcss-selector-parser"
+import {
+  report,
+  ruleMessages,
+  styleSearch,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-attribute-square-brackets-space-inside"
+
+export const messages = ruleMessages(ruleName, {
+  expectedOpening: "Expected single space after \"[\"",
+  rejectedOpening: "Unexpected whitespace after \"[\"",
+  expectedClosing: "Expected single space before \"]\"",
+  rejectedClosing: "Unexpected whitespace before \"]\"",
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      selectorParser(selectorTree => {
```

Comment:
I'd say let's run one quick optimization here: if there is no `[` in the selector, don't bother parsing it. 

Does that seem right?